NGitLab/IProjectClient.cs

```diff
@@ -56,6 +56,8 @@ public interface IProjectClient
 
         Task<Project> GetByIdAsync(int id, SingleProjectQuery query, CancellationToken cancellationToken = default);
 
+        Task<Project> GetByFullNameAsync(string fullName, SingleProjectQuery query = null, CancellationToken cancellationToken = default);
```

Comment:
I think we should use `GetByIdAsync` or `GetByPathWithNamespaceAsync`. FullName is confusing as it refers to the display name which could be different from the path.

Reply:
Maybe we should create a struct ProjectIdOrPathWithNamespace with implicit converters from string and int. This way have only one overload (not for this method as the other method already exists, but it could be useful for the following methods).

Reply:
Based on terminology found in the [GitLab doc](https://docs.gitlab.com/ee/api/index.html#namespaced-path-encoding), I went ahead with `GetByNamespacedPathAsync`.

Reply:
Having said that, I agree that
1. "Full Name" was incorrect... *though that's the name of the `ProjectClient` indexer's parameter* 
2. It makes sense to eventually introduce some kind of `ProjectIdOrPath` type